freqtrade/persistence/migrations.py

```diff
@@ -167,6 +168,13 @@ def check_migrate(engine, decl_base, previous_tables) -> None:
         inspector = inspect(engine)
         cols = inspector.get_columns('trades')
 
+    if not has_column(cols, 'buy_tag'):
+        logger.info(f'Running database migration for trades - backup: {table_back_name}')
+        migrate_trades_table(decl_base, inspector, engine, table_back_name, cols)
+        # Reread columns - the above recreated the table!
+        inspector = inspect(engine)
+        cols = inspector.get_columns('trades')
+
```

Comment:
This is wrong and kinda dangerous (if you're comming from an older version, migrations might run twice).

I've fixed this now - please look at that commit to see how easy this would've been :D